variables.tf

```
@@ -136,3 +142,9 @@ variable "root_block_device_encrypted" {
   default     = true
   description = "Whether to encrypt the root block device"
 }
+
+variable "nat_instance_public_ssh_keys" {
+  type        = list(string)
+  description = "SSH public keys that we want to attach to the nat instance"
+  default     = []
+}
```

I don't think we should be allowing manual ssh keys as an input.

You could expose user_data instead and pass in the keys manually if you like, but this is unnecessary if ssm access is enabled

```suggestion
variable "nat_instance_public_ssh_keys" {
  type        = list(string)
  description = "SSH public keys that we want to attach to the nat instance"
  default     = []
}
```